addon/-private/system/model/internal-model.js

```diff
@@ -119,7 +119,10 @@ export default class InternalModel {
   constructor(modelName, id, store, data) {
     heimdall.increment(new_InternalModel);
     this.id = id;
-    this._internalId = InternalModelReferenceId++;
+
+    // this ensure ordered set can quickly identify this as unique
+    this[Ember.GUID_KEY] = InternalModelReferenceId++ + 'internal-model';
```

Comment:
The comment will help this survive a lot better, the assumption before was that this shaping was to help the cases where internalModel was being abused by DS.Model (no longer is), meta, computeds and other public interface things.